packages/solidity-contracts/deploy/beta5/005.fuel_message_portal_v2.ts

```diff
@@ -0,0 +1,40 @@
+import type { HardhatRuntimeEnvironment } from 'hardhat/types';
+import type { DeployFunction } from 'hardhat-deploy/dist/types';
+
+import { FuelMessagePortalV2__factory as FuelMessagePortalV2 } from '../../typechain';
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const {
+    ethers,
+    upgrades: { upgradeProxy, erc1967 },
+    deployments: { get, save },
+  } = hre;
+  const [deployer] = await ethers.getSigners();
+
+  const fuelMessagePortal = await get('FuelMessagePortal');
+
+  const { deployTransaction, address } = await upgradeProxy(
+    fuelMessagePortal,
+    new FuelMessagePortalV2(deployer),
+    {
+      unsafeAllow: ['constructor'],
+      constructorArgs: [ethers.constants.MaxUint256],
+    }
+  );
+
+  await deployTransaction.wait();
+  const implementation = await erc1967.getImplementationAddress(address);
```

Comment:
Haven't seen this erc before. Can I get a short summary?

Reply:
It is part of OZ upgrades plugin. It has always been there, in our handmade scripts too.

Just an standard for proxy contracts (TransparentUpgradableProxy, UUPS, etc).